spec/porky_lib/file_service_spec.rb

```diff
@@ -94,227 +95,235 @@ def write_test_file(data)
-      file_service.overwrite_file(File.read("spec#{File::SEPARATOR}porky_lib#{File::SEPARATOR}data#{File::SEPARATOR}large_plaintext"), default_file_key,
-                                  bucket_name, default_key_id)
-    end.to raise_exception(PorkyLib::FileService::FileSizeTooLargeError)
-  end
+    it 'attempt to write with key ID nil raises FileServiceError' do
+      expect do
+        file_service.write(plaintext_data, bucket_name, nil)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
 
-  it 'read encrypted data from S3' do
-    file_key = file_service.write(plaintext_data, bucket_name, default_key_id)
+    it 'attempt to write to bucket without permission raises FileServiceError' do
+      Aws.config[:s3].delete(:stub_responses)
+      Aws.config[:s3] = {
+        stub_responses: {
+          put_object: 'Forbidden'
+        }
+      }
+      expect do
+        file_service.write(plaintext_data, bucket_name, default_key_id)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
 
-    plaintext, should_reencrypt = file_service.read(bucket_name, file_key)
-    expect(plaintext_data).to eq(plaintext)
-    expect(should_reencrypt).to be_falsey
+    it 'attempt to write to bucket that does not exist raises FileServiceError' do
+      Aws.config[:s3].delete(:stub_responses)
+      Aws.config[:s3] = {
+        stub_responses: {
+          put_object: 'NotFound'
+        }
+      }
+      expect do
+        file_service.write(plaintext_data, bucket_name, default_key_id)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
   end
 
-  it 'read encrypted data from S3 with directory' do
-    dir_name = 'directory1/dirA'
-    file_key = file_service.write(plaintext_data, bucket_name, default_key_id, directory: dir_name)
-    expect(file_key).to include(dir_name)
+  describe '#overwrite_file' do
+    it 'overwrite encrypted data to S3' do
+      expect do
+        file_service.overwrite_file(plaintext_data, default_file_key, bucket_name, default_key_id)
+      end.not_to raise_exception
+    end
 
-    plaintext, should_reencrypt = file_service.read(bucket_name, file_key)
-    expect(plaintext_data).to eq(plaintext)
-    expect(should_reencrypt).to be_falsey
-  end
+    it 'overwrite large encrypted data to S3' do
+      expect do
+        file_service.overwrite_file(plaintext_data_large, default_file_key,
+                                    bucket_name, default_key_id)
+      end.not_to raise_exception
+    end
 
-  it 'read encrypted data from S3 which should be re-encrypted' do
-    stub_data_to_be_reencrypted
-    file_key = file_service.write(plaintext_data, bucket_name, default_key_id)
+    it 'overwrite encrypted data to S3 with metadata' do
+      expect do
+        metadata = { content_type: 'test/data' }
+        file_service.overwrite_file(plaintext_data, default_file_key, bucket_name, default_key_id, metadata: metadata)
+      end.not_to raise_exception
```

Comment:
For these, should we not be testing to ensure the correct AWS S3 API call gets made with the correct parameters?